bluesky/bundlers.py

```diff
@@ -196,18 +222,34 @@ async def read(self, msg, reading):
             # Ask the object for any resource or datum documents is has cached
             # and cache them as well. Likewise, these will be emitted if and
             # when _save is called.
-            if hasattr(obj, "collect_asset_docs"):
-                self._asset_docs_cache.extend(
-                    obj.collect_asset_docs(*msg.args, **msg.kwargs)
-                )
+            self._asset_docs_cache.extend(
+                maybe_collect_asset_docs(obj, *msg.args, **msg.kwargs)
+            )
 
         return reading
 
-    def _cache_config(self, obj):
+    async def _cache_describe(self, obj):
+        "Read the object's describe and cache it."
+        obj = check_supports(obj, Readable)
+        self._describe_cache[obj] = await maybe_await(obj.describe())
+
+    async def _cache_describe_config(self, obj):
+        "Read the object's describe_configuration and cache it."
+        if isinstance(obj, Configurable):
```

Comment:
`read_configuration()` and `describe_configuration()` will only be called if they both exist, as `Configurable` has both methods. This will get worse if we add `configure()` to `Configurable`. We will need an Ability name for `validate()` too

Reply:
Resolved as we ditched `configure()` in favour of `set()` on the relevant `Signal`s, and it is reasonable to expect both `read_configuration()` and `describe_configuration()` to be implemented